src/switch_ivr_play_say.c

```diff
@@ -1781,43 +1789,92 @@ SWITCH_DECLARE(switch_status_t) switch_ivr_play_file(switch_core_session_t *sess
 
 			if (!switch_test_flag(fh, SWITCH_FILE_NATIVE) && fh->speed && do_speed) {
 				float factor = 0.25f * abs(fh->speed);
-				switch_size_t newlen, supplement, step;
-				short *bp = write_frame.data;
-				switch_size_t wrote = 0;
-
-				supplement = (int) (factor * olen);
-				if (!supplement) {
-					supplement = 1;
+				short* bp = write_frame.data;
+				switch_size_t supplement = (int) (factor * olen);
+				switch_size_t newlen = (fh->speed > 0) ? olen - supplement : olen + supplement;
+				int src_rng = (fh->speed > 0 ? supplement : sp_cut_src_rng)* sp_has_overlap;
+				int datalen = newlen + src_rng + sp_fadeLen;
+				int extra = datalen - olen;
+				short* data = NULL;
+				short* currp = NULL;
+				switch_zmalloc(data, datalen * 2);
```

Comment:
malloc on every frame seems wasteful

Reply:
heap buf max sized with realloc or a static buf sized to SWITCH_RECOMMENDED_BUFFER_SIZE which is designed to fit sane data frames.

Reply:
@rucc  looks like you missed on removing the `switch_zmalloc` and the `data` variable, it is not needed anymore.

Reply:
@ar45 Thanks for your suggestions... data is still needed as a buffer but zmalloc is not needed.
https://github.com/yois615/freeswitch/commit/96ad56073d8c05149747770e869988a3e765e15e

I will open a PR against @rucc branch